drivers/serial/uart_ns16550.c

```diff
@@ -1964,6 +1985,14 @@ static DEVICE_API(uart, uart_ns16550_driver_api) = {
 			(.dlf = DT_INST_PROP_OR(n, dlf, 0),))			     \
 		DEV_DATA_ASYNC(n)						     \
 
+#if defined(CONFIG_UART_NS16550_ENABLE_DEVICE_PM)
+#define PM_DEVICE_INST_DEFINE_NS16550	PM_DEVICE_DT_INST_DEFINE(n, uart_ns16550_pm_action);
+#define PM_DEVICE_INST_GET_NS16550      PM_DEVICE_DT_INST_GET(n)
+#else
+#define PM_DEVICE_INST_DEFINE_NS16550
+#define PM_DEVICE_INST_GET_NS16550      NULL
+#endif
```

Comment:
`PM_DEVICE_DT_INST_DEFINE()` and `PM_DEVICE_DT_INST_GET()` are already no-op if `CONFIG_PM_DEVICE=n` (and `CONFIG_UART_NS16550_ENABLE_DEVICE_PM` depends on `CONFIG_PM_DEVICE`)